go/app/main.go

```diff
@@ -37,78 +38,134 @@ func root(c echo.Context) error {
 	return c.JSON(http.StatusOK, res)
 }
 
-func getFileContent(c echo.Context) (map[string][]Item, error) {
-	// Get data from items.json
-	f, err := os.OpenFile("items.json", os.O_RDONLY|os.O_CREATE, os.ModePerm)
+func getData(c echo.Context) (*sql.Rows, error) {
+	// Get all data form database
+	db, err := sql.Open("sqlite3", "../db/mercari.sqlite3")
 	if err != nil {
 		c.Logger().Errorf("err: %v", err)
-		return nil, err
+		res := Response{Message: "Fail to open database"}
+		return nil, c.JSON(http.StatusInternalServerError, res)
 	}
-	defer f.Close()
-
-	bytes, err := ioutil.ReadAll(f)
+	defer db.Close()
+	rows, err := db.Query(
+		`SELECT * FROM items inner join category on items.category_id = category.id `,
+	)
 	if err != nil {
 		c.Logger().Errorf("err: %v", err)
-		return nil, err
-	}
-	items := make(map[string][]Item)
-	if len(bytes) > 0 {
-		err = json.Unmarshal(bytes, &items)
-		if err != nil {
-			c.Logger().Errorf("err: %v", err)
-			return nil, err
-		}
+		res := Response{Message: "Query is not valid"}
+		return nil, c.JSON(http.StatusInternalServerError, res)
 	}
-	return items, nil
+	return rows, nil
 }
 
 func imageHash(image *multipart.FileHeader, c echo.Context) ([]byte, error) {
 	imagePath := path.Join(ImgDir, image.Filename)
 	imageFile, err := os.Open(imagePath)
 	if err != nil {
 		c.Logger().Errorf("err: %v", err)
-		return nil, err
+		res := Response{Message: "Image is not valid"}
+		return nil, c.JSON(http.StatusBadRequest, res)
 	}
 	defer imageFile.Close()
 
 	// Hashing image file
 	hash := sha256.New()
 	if _, err := io.Copy(hash, imageFile); err != nil {
 		c.Logger().Errorf("err: %v", err)
-		return nil, err
+		res := Response{Message: "Fail to hash image"}
+		return nil, c.JSON(http.StatusInternalServerError, res)
 	}
 	hashedImage := hash.Sum(nil)
 
 	c.Logger().Infof("Receive image: %x.jpg", hashedImage)
 	return hashedImage, nil
 }
 
-func writeItems(item Item, c echo.Context) error {
-	// Add item to item.json
-	items, err := getFileContent(c)
+func searchCategory(cat string, c echo.Context) (int, error) {
+	// Find out if a category exists
+	db, err := sql.Open("sqlite3", "../db/mercari.sqlite3")
 	if err != nil {
 		c.Logger().Errorf("err: %v", err)
-		return err
+		res := Response{Message: "Fail to open database"}
+		return -1, c.JSON(http.StatusInternalServerError, res)
 	}
-	// items := make(map[string][]Item)
-	items["items"] = append(items["items"], item)
-	jsonData, err := json.Marshal(items)
+	defer db.Close()
+
+	rows, err := db.Query(
+		`SELECT * FROM category`,
+	)
 	if err != nil {
 		c.Logger().Errorf("err: %v", err)
-		return err
+		res := Response{Message: "Query is not valid"}
+		return -1, c.JSON(http.StatusInternalServerError, res)
```

Comment:
[nits] 関数が冗長に感じるので、resは直接埋め込んでしまってもいいような気がします。他の部分も同様です。
```suggestion
		return -1, c.JSON(http.StatusInternalServerError, Response{Message: "Query is not valid"})
```